docs/extending-with-custom-configurable-dsl-components.md

```diff
@@ -71,8 +71,11 @@ data class GradleBuildDsl(
 ) {
     ...
     fun DslContext<Step>.gradleCommand(command: String, additionalBuildArgs: Var.Literal.Str) =
-            withEnv(mapOf("GRADLE_USER_HOME" to "${"WORKSPACE".environmentVar()}/.gradle-home-tmp")) { artifactoryAuthenticated {
-                sh(("./gradlew --no-daemon --stacktrace --build-cache " +
+            withEnv(
+                mapOf("GRADLE_USER_HOME" to "${"WORKSPACE".environmentVar()}/.gradle-home-tmp",
+                "JENKINS_NODE_COOKIE" to "dontKillMe")
```

Comment:
This is desired per https://stackoverflow.com/a/39169504.
So TL;DR Jenkins has a process tree killer that keeps track of what processes it starts so it can kill them when the stage ends as cleanup. This is problematic for the gradle daemon though as that can be shared across stages so if it gets killed it could impact other builds if they are running with the daemon on the same node.